front/deployfile_download.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ * GLPI Inventory Plugin
+ * Copyright (C) 2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on FusionInventory for GLPI
+ * Copyright (C) 2010-2021 by the FusionInventory Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI Inventory Plugin.
+ *
+ * GLPI Inventory Plugin is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI Inventory Plugin is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with GLPI Inventory Plugin. If not, see <https://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+use org\bovigo\vfs\FilenameTestCase;
+
+include("../../../inc/includes.php");
+
+Session::checkRight('plugin_glpiinventory_package', READ);
+global $CFG_GLPI;
+
+$filename = null;
+if (isset($_GET['filename'])) {
+    $filename = $_GET['filename'];
+}
+
+$mimetype = null;
+if (isset($_GET['mimetype'])) {
+    $mimetype = urldecode($_GET['mimetype']);
+}
+
+if (isset($_GET['sha512'])) {
+   $deploy = new PluginGlpiinventoryDeployFile();
+   if($deploy->checkPresenceFile($_GET['sha512'])) {
+      $path = $deploy->getFilePath($_GET['sha512']);
+      $deploy->constructFileToTmp($path, $filename);
```

Comment:
Merging file parts into a temporary file to then read it will consume lots of memory.

Maybe using `Toolbox::sendFile` is not the best approach here. A dedicated logic coul be used.

Instead of writing a temporary file, maybe it could be directly send to output, using this kind of logic:
```php
$stdout = fopen('php://output', 'w');
foreach ($path as $key => $value) {
    $part = gzopen($value, 'r');
    fwrite($stdout, gzread($part, 1024 * 1024));
    gzclose($part);
}
fclose($stdout);
```

Some tests should be done to see what is the best approach to limit memory usage.